apps/src/lab2/levelEditors/validations/EditValidation.tsx

```diff
@@ -50,126 +52,173 @@ const EditValidation: React.FunctionComponent<EditValidationProps> = ({
+            placeholder="Optional comment..."
+            value={validation.comment}
+            onChange={e => {
+              validation.comment = e.target.value;
+              onValidationChange(validation);
+            }}
+          />
+        </div>
+        <div className={moduleStyles.column}>
+          <div className={moduleStyles.buttonsRow}>
+            <button
+              type="button"
+              onClick={() => moveValidation(validation.key, 'up')}
+              className={moduleStyles.moveValidationButton}
+            >
+              <FontAwesome icon="arrow-up" title={undefined} className="icon" />
+            </button>
+            <button
+              type="button"
+              onClick={() => moveValidation(validation.key, 'down')}
+              className={moduleStyles.moveValidationButton}
+            >
+              <FontAwesome
+                icon="arrow-down"
+                title={undefined}
+                className="icon"
+              />
+            </button>
+            <button
+              type="button"
+              onClick={() => duplicateValidation(validation)}
+              className={moduleStyles.duplicateValidationButton}
+            >
+              <FontAwesome icon="copy" title={undefined} className="icon" />
+            </button>
+            <button
+              type="button"
+              onClick={() => deleteValidation(validation.key)}
+              className={moduleStyles.deleteValidationButton}
+            >
+              <FontAwesome icon="trash" title={undefined} className="icon" />
+            </button>
+          </div>
+        </div>
       </div>
-      <div className={moduleStyles.row}>
-        <label htmlFor="next" className={moduleStyles.label}>
-          Passes Level?
-        </label>
-        <input
-          type="checkbox"
-          id="next"
-          name="next"
-          checked={validation.next}
-          onChange={e => {
-            validation.next = e.target.checked;
-            onValidationChange(validation);
-          }}
-        />
+      <div className={moduleStyles.section}>
+        <div
+          className={classNames(moduleStyles.column, moduleStyles.columnFirst)}
+        >
+          {'When'}
+        </div>
+        <div className={moduleStyles.column}>
+          {validation.conditions.map((condition, index) => {
+            return (
+              <div className={moduleStyles.row} key={`${index}-condition`}>
+                <EditCondition
+                  condition={condition}
+                  conditionTypes={conditionTypes}
+                  index={index}
+                  onConditionChange={onConditionChange}
+                  deleteCondition={deleteCondition}
+                />
+                <div
+                  className={moduleStyles.row}
+                  key={`${index}-description`}
+                />
+                {index === validation.conditions.length - 1 && (
+                  <button
```

Comment:
Optional for this PR, but thoughts on replacing these with DSCO buttons now that we have them?